appcelerator/titanium_mobile
# [TIMOB-26294] TiAPI: Notify app about user-interactions

**JIRA**: https://jira.appcelerator.org/browse/TIMOB-26294

In order to create a solution for https://jira.appcelerator.org/browse/TIMOB-13884, we're trying to capture all user interaction events. For Android, there is support to capture all onUserInteraction events. (https://jira.appcelerator.org/browse/TIMOB-26278)

We want to achieve the same functionality for iOS by extending the UIApplication and trigger an event for each UIEvent. In order to avoid negative performance impact, an event will be triggered only when a touch event began.

**Test:**
Run the below test on both Android and iOS.
1. Set up a project to use the `UserInteractionAutoCloseTest.js` attached to [TIMOB-26294](https://jira.appcelerator.org/browse/TIMOB-26294).
2. Build and run the app onto Android/iOS.
3. Tap on the "Open Child Window" button.
4. Note the "Will auto-close" countdown text at the top. (It starts from 10 seconds.)
5. Wait for the countdown to reach zero and for the child window to close.
6. Tap on the "Open Child Window" button.
7. Tap on an empty part of the window. (Verify countdown resets.)
8. Tap on the countdown text. (Verify countdown resets.)
9. Press the device's volume up/down buttons. (Verify countdown resets.)
10. Tap on the TextField.
11. Verify that countdown resets every time you enter/delete characters.
12. Tap on the "Show Alert" button.
13. On iOS, verify tapping outside of alert resets the countdown. (Not supported on Android.)
